app/controllers/api/v2/courses_controller.rb

```diff
@@ -125,15 +121,11 @@ def create
         authorize @provider, :can_create_course?
         return unless course_params.values.any?
 
+        build_new_course
         course_code = @provider.next_available_course_code
-        @course = Course.new(provider: @provider)
-        @course.assign_attributes(course_params.merge(course_code: course_code))
-        update_subjects
-        update_sites
-        update_further_education_fields if @course.level == "further_education"
-        @course.name = @course.generate_name
+        @course.assign_attributes(course_code: course_code)
 
-        if @course.save
+        if @course.valid?(:new) && @course.save
```

Comment:
Shouldn't `.save()` also do validation?

Reply:
It _does_ - but not with the scope `:new`, only with the scope `:create`